packages/launchpad/cypress/e2e/migration.cy.ts

```diff
@@ -743,6 +743,61 @@ describe('Full migration flow for each project', { retries: { openMode: 2, runMo
     migrateAndVerifyConfig()
   })
 
+  it('completes journey for migration-e2e-duplicated-spec-names', () => {
+    startMigrationFor('migration-e2e-duplicated-spec-names')
+    // default testFiles - auto
+    cy.get(renameAutoStep).should('exist')
+    cy.get(configFileStep).should('exist')
+
+    // Migration workflow
+    // before auto migration
+    cy.contains('cypress/integration/app-spec2.js')
+    cy.contains('cypress/integration/app_spec2.js')
+    cy.contains('cypress/integration/app.spec.js')
+    cy.contains('cypress/integration/app2_spec.js')
+    cy.contains('cypress/integration/app_spec.js')
+    cy.contains('cypress/integration/app-spec.js')
+
+    // after auto migration
+    cy.contains('cypress/e2e/app-spec2.cy.js')
+    cy.contains('cypress/e2e/app_spec2.cy.js')
+    cy.contains('cypress/e2e/app.cy.js')
+    cy.contains('cypress/e2e/app2.cy.js')
+    cy.contains('cypress/e2e/app_spec.cy.js')
+    cy.contains('cypress/e2e/app-spec.cy.js')
```

Comment:
For these assertions, it's worth asserting the render order to ensure that the before and after names are mapped appropriately. Otherwise, I don't think your sorted extension prioritization has coverage.

```js
// these test ids don't currently exist
cy.get('[data-cy="before-file"]').eq(0).should('contain', 'cypress/integration/app-spec2.js')
cy.get('[data-cy="after-file"]').eq(0).should('contain', 'cypress/e2e/app-spec2.cy.js')
```